composer.json

```diff
@@ -11,16 +11,17 @@
         }
     ],
     "require": {
-        "php": "^5.4|7.*",
-        "puli/composer-plugin": "1.0.0-beta9"
+        "php": "^5.4|7.*"
     },
     "require-dev": {
         "php-http/httplug": "^1.0",
         "php-http/message-factory": "^1.0",
+        "puli/composer-plugin": "1.0.0-beta9",
         "phpspec/phpspec": "^2.4",
         "henrikbjorn/phpspec-code-coverage" : "^1.0"
     },
     "suggest": {
+        "puli/composer-plugin": "Allow to use Discovery",
```

Comment:
still fail to see the point, sorry. i could do
```
if (!$client) {
    if (!class_exists('Http\Discovery...')) {
        throw new Exception('Specify client or include discovery');
    }
    $client = ClientDiscovery::find();
}
```

making puli optional in discovery does not change anything here, except that including discovery in a project becomes more tricky.